loader/fixups.c

```diff
@@ -39,7 +39,30 @@ SYS_INIT(disable_bootloader_mpu, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAU
 SYS_INIT(disable_mpu_rasr_xn, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
 #endif
 
-#if defined(CONFIG_BOARD_ARDUINO_GIGA_R1) && defined(CONFIG_VIDEO)
+#if defined(CONFIG_INPUT)
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+	
+typedef void (*zephyr_input_callback_t)(struct input_event *evt, void *user_data);
+
+static zephyr_input_callback_t zephyr_input_cb = NULL;
+
+void zephyr_input_register_callback(zephyr_input_callback_t cb) {
+	zephyr_input_cb = cb;
+}
+
+static void zephyr_input_callback(struct input_event *evt, void *user_data) {
+	if (zephyr_input_cb) {
+		zephyr_input_cb(evt, user_data);
+    }
+}
+
+INPUT_CALLBACK_DEFINE(NULL, zephyr_input_callback, NULL);
+#endif
+
+#if (defined(CONFIG_BOARD_ARDUINO_GIGA_R1) || defined(CONFIG_BOARD_ARDUINO_PORTENTA_H7)) \
+		&& defined(CONFIG_VIDEO)
```

Comment:
```suggestion
#if defined(CONFIG_BOARD_ARDUINO_GIGA_R1) && defined(CONFIG_VIDEO)
```
Sorry, I missed this. This enables the camera clock for Portenta. Whether it's needed or not, it doesn't belong to this commit/PR.